### PR TITLE
feat: Improved custom namespace options

### LINKED ISSUE #676: [Helm] Improve custom namespace options
The Helm chart allows the definition of a [custom default namespace](https://github.com/open-feature/open-feature-operator/blob/339e5c8cf49c54660ed3ee70fe988345b55de999/chart/open-feature-operator/README.md?plain=1#L98) where all OFO's resources will be created. Some improvements are needed here to support more use-cases.

We should:

- Add the ability to customize the namespace in use by OFO, _without actually creating it_.
- Add the ability to add additional labels and annotations to the created namespace.

### PR DESCRIPTION
**Pull request by 7h3-3mp7y-m4n**
## This PR
- Added the Custom name flexibility
- Add the ability to add additional labels and annotations to the created namespace.

### Related Issues

Fixes #676

### CODE REVIEW

File: chart/open-feature-operator/values.yaml [hunk 1]
```diff
@@ -2,6 +2,11 @@
 ## @section Global
 ## @param  defaultNamespace To override the namespace use the `--namespace` flag. This default is provided to ensure that the kustomize build charts in `/templates` deploy correctly when no `namespace` is provided via the `-n` flag.
 defaultNamespace: open-feature-operator-system
+namespace:
+  name: open-feature-operator-system
+  create: true
+  labels: {}
+  annotations: {}
```
Comment (toddbaert) [inline chart/open-feature-operator/values.yaml:9]: How does this work in conjunction with the old `defaultNamespace`? What happens if you set `defaultNamespace` to something different than `namespace.name`? Should we remove `defaultNamespace`?

**Comment by cpitstick-latai** [comment]
Please update the readme as well.

**Comment by toddbaert** [comment]
I'd like this to resolve https://github.com/open-feature/open-feature-operator/issues/683 as well.

@7h3-3mp7y-m4n are you still planning on working on this? No pressure! Just want to know your intent!

**Comment by 7h3-3mp7y-m4n** [comment]
okay @toddbaert I'll defenitly like to work on this and let you know on Slack.

**Comment by cpitstick-latai** [comment]
I've created a variation of this here to unblock progress on this issue: https://github.com/open-feature/open-feature-operator/pull/702

**Comment by beeme1mr** [comment]
Closing this PR in favor of https://github.com/open-feature/open-feature-operator/pull/702

**Pull request closed without merging**